modules/ppcp-wc-gateway/src/Gateway/PayUponInvoice/PayUponInvoice.php

```diff
@@ -270,6 +269,32 @@ function( array $methods ): array {
 		);
 	}
 
+	/**
+	 * Checks whether checkout is ready for PUI.
+	 *
+	 * @return bool
+	 */
+	private function is_checkout_ready_for_pui(): bool {
+		$billing_country = filter_input( INPUT_POST, 'country', FILTER_SANITIZE_STRING ) ?? null;
+		if ( $billing_country && 'DE' !== $billing_country ) {
+			return false;
+		}
+
+		if ( 'EUR' !== get_woocommerce_currency() ) {
+			return false;
+		}
+
+		$cart = WC()->cart ?? null;
+		if ( $cart ) {
+			$cart_total = (int) WC()->cart->get_totals()['total'];
```

Comment:
I used `(float) ->get_total( 'numeric' )` instead of `get_totals` somewhere else, not sure if there is any difference.